allennlp/data/iterators/data_iterator.py

```diff
@@ -53,5 +53,5 @@ def from_params(cls, params: Params):
         from . import iterators
         # TODO(Mark): The adaptive iterator will need a bit of work here,
         # to retrieve the scaling function etc.
-        iterator_type = params.pop_choice("type", iterators.keys())
-        return iterators[iterator_type](**params.as_dict())
+        iterator_type = params.pop_choice("type", list(iterators.keys()))
+        return iterators[iterator_type](**params.as_dict())  # type: ignore
```

Comment:
it looks like it's because the base class doesn't have an `__init__` method, if I define one that takes `**kwargs` and does nothing it passes